dsgrid/query/query_context.py

```diff
@@ -68,7 +69,18 @@ def _consolidate_pivoted_metadata(self):
                     "Please set the output format to 'unpivoted'."
                 )
 
-    def get_pivoted_columns(self, dataset_id=None):
+    def get_value_columns(self) -> set[str]:
+        """Return the value columns in the final dataset."""
+        match self.get_table_format_type():
+            case TableFormatType.PIVOTED:
+                return self.get_pivoted_columns()
```

Comment:
Looks like get_pivoted_columns() uses get_dimension_column_names(), which uses _dataset_metadata(). I didn't trace this all the way down but can I assume the _dataset_metadata() stays up to date with what's available from the dataset, like if the dataset has been filtered?

Reply:
yes